scipy/sparse/linalg/_isolve/minres.py

```diff
@@ -312,9 +313,7 @@ def minres(A, b, x0=None, shift=0.0, tol=1e-5, maxiter=None,
                 istop = 3
             # if rnorm <= epsx   : istop = 2
             # if rnorm <= epsr   : istop = 1
-            if test2 <= tol:
-                istop = 2
-            if test1 <= tol:
+            if testrnorm <= tol:
                 istop = 1
```

Comment:
Could you please explain why the `test2 <= tol` branch was no longer necessary? And it would seem that the first comment here would also be unnecessary then.

Reply:
In fact, I think we should use the standard `||r||/||b||` instead of `||r||/||A||||x||` and `||Ar||/||A||||r||` as the criterion of iterative convergence, and it is not expensive to calculate `||r||/||b||`. So I use `testrnorm <= tol` on line 316

Reply:
`||r||/||b|| < tol` serves as a standard convergence criterion. CG, Bi-CGStab, GMRES in SciPy are using `||r||/||b|| < tol` (It is expected to have a consistent convergence criterion without increasing the amount of calculation) and this criterion is also used in some known math libraries, such as PETSc, PyAMG etc.

Reply:
@adeak what's your opinion on the state of this PR?

Reply:
@tylerjereddy I'm afraid I don't have the domain knowledge nor the familiarity with SciPy to be able to review the PR properly. The fact that a flag like `istop` can seemingly change in a float -> complex enhancement stood out to me, but the author's reply was technical enough that it needs more expertise than what I have.

The rest of the changes looked good to me, for what it's worth.